### PR TITLE
Rebase #384 onto master

### PR DESCRIPTION
**Pull request by Varunram**
https://github.com/mit-dci/lit/pull/384 was a PR that couldn't be added to the other list because of a merge conflict. This is #384 rebased unto current master. @delbonis please verify if this contains all changes. If not, you can push to this fork directly to implement changes.

### CODE REVIEW

**Comment by delbonis** [comment]
If it builds properly then I think it should have all of the important changes.

File: test/tests.txt [hunk 1]
```diff
@@ -10,13 +14,13 @@ send2 2
 
 # Simple channel operations
 fund 2
-close 2 run_test_forward
-close 2 run_test_reverse
+close 2 forward
+close 2 reverse
 # Disabled these to tests until we support breaking right after funding (we don't)
```
Comment (delbonis) [inline test/tests.txt]: We can remove this line.

**Comment by Varunram** [comment]
it build fine and has the same diff, but I didn't test it much after that.  Test failure is due to the env not being initiated, so will rebuild.

**Comment by delbonis** [comment]
Should be all ok then.

**Pull request merged**